### PR TITLE
Jest emotion/improvements

### PR DESCRIPTION
**Pull request by Andarist**
Match rules in declarations with component used as a selector and take specificity into account when matching styles

fixes https://github.com/emotion-js/emotion/issues/1214

### CODE REVIEW

**Comment by FezVrasta** [comment]
Thanks!

File: yarn.lock [hunk 1]
```diff
@@ -1282,6 +1282,16 @@
     exec-sh "^0.3.2"
     minimist "^1.2.0"
 
+"@emotion/babel-preset-css-prop@^10.0.5":
```
Comment (Andarist) [inline yarn.lock]: im not sure why our babel plugins were added to the lock 🤔

Comment (FezVrasta) [reply yarn.lock]: It usually happens if some dependency is referencing a version different from latest, or when yarn shits its pants (it happens often)

Comment (Andarist) [reply yarn.lock]: actually I've found why this has happened right now with a help of `yarn why` - those are dependencies of `gatsby-plugin-emotion`, so it makes sense that they have been added

**Pull request merged**